From 2c7030f67d31dec494f9831f8250ae3da284ace3 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 26 Mar 2021 23:15:24 +0000
Subject: [PATCH] gnu: Add rtg-math

* gnu/packages/lisp-xyz.scm: (sbcl-rtg-math, cl-rtg-math, ecl-rtg-math): New variables
---
 gnu/packages/lisp-xyz.scm | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index d4bf75b231..562113366a 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -2890,6 +2890,39 @@ from GLSL as data.")
 (define-public cl-glsl-spec
   (sbcl-package->cl-source-package sbcl-glsl-spec))
 
+(define-public sbcl-rtg-math
+  (let ((commit "29fc5b3d0028a4a11a82355ecc8cca62662c69e0")
+        (revision "1"))
+    (package
+      (name "sbcl-rtg-math")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/cbaggers/rtg-math")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "0bhxxnv7ldkkb18zdxyz2rj2a3iawzq2kcp7cn5i91iby7n0082x"))))
+      (build-system asdf-build-system/sbcl)
+      (inputs
+       `(("alexandria" ,sbcl-alexandria)
+         ("glsl-symbols" ,sbcl-glsl-spec)
+         ("documentation-utils" ,sbcl-documentation-utils)))
+      (home-page "https://github.com/cbaggers/rtg-math")
+      (synopsis "Common Lisp library providing common math functions used in games")
+      (description
+       "RTG-MATH provides a selection of the math routines most commonly needed
+for making realtime graphics in lisp.")
+      (license license:public-domain))))
+
+(define-public ecl-rtg-math
+  (sbcl-package->ecl-package sbcl-rtg-math))
+
+(define-public cl-rtg-math
+  (sbcl-package->cl-source-package sbcl-rtg-math))
+
 (define-public sbcl-varjo
   (let ((commit "9e77f30220053155d2ef8870ceba157f75e538d4")
         (revision "1"))
-- 
2.29.2

